pinc/graph_data.inc

```diff
@@ -857,3 +838,16 @@ function query_graph_cache($function, $args = [], $expiration = 3600)
 {
     return memoize_function($function, $args, $expiration, get_desired_language());
 }
+
+/**
+ * Given an array of numbers and a window, calculate the moving average over
+ * the window.
+ */
+function calculate_sma(array $array, int $period): array
+{
+    $results = [];
```

Comment:
This is quite an inefficient way of doing this: you're doing `N` `array_slice`s and `N*period` additions to get your SMA. A better way is to calculate the sum / $period to give you the first result, and then for each subsequent result subtract off the oldest element (divided by $period) and add the newest element (divided by $period).

(That said, we should probably profile the code before doing optimizations like this.)

Reply:
See aforementioned "this is already stupid fast" comment.